docs/modules/ROOT/pages/explanations/decisions/admin-kubeconfig.adoc

```diff
@@ -0,0 +1,99 @@
+
+We've got two options to generate tokens for service accounts:
+
+. The https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount[TokenRequest] API allows us to generate service account tokens which expire after a defined amount of time.
+However, tokens which are manually created through the TokenRequest API (for example with `kubectl create token`) can't be invalidated before they expire.
+
+. [Non-expiring API tokens] are created by defining a secret of type `kubernetes.io/service-account-token`.
+As the name suggests, these tokens don't expire.
+
+The only way to permanently invalidate service account tokens (both non-expiring and time-bound) is to delete the service account.
+Creating a new service account with the same name in the same namespace doesn't reactivate tokens associated with a previous service account, since the tokens contain the service account's Kubernetes resource UID.
+
+The proposed approach for using service account tokens is to use the TokenRequest API to create short-lived API tokens to generate expiring admin credentials by default.
+Additionally, introduce a mechanism to force the tool to recreate the service account to invalidate any old tokens that might have leaked.
+That mechanism might be as simple as having the tool reconcile the service account and recreate it if it gets deleted.
+
+=== Credential management
+
+==== Extend Steward to manage credentials and write them to Vault
+
+We can extend https://syn.tools/steward[Steward] to manage and renew the credentials and store them in Vault.
+
+This allows us to issue relatively short-lived credentials (on the order of days), which limits the attack surface presented by engineers accessing admin credentials in emergency situations.
+
+Optionally, we can also extend Steward to render a full kubeconfig file based on the managed credentials and store that file in Vault in addition to the raw credentials.
+If we store a full kubeconfig file in Vault, we can document a single `vault` CLI command which fetches the emergency kubeconfig for a cluster.
+
+==== Create a new tool which manages the credentials and writes them to Vault
+
+Instead of extending Steward, we could also create a new tool which manages admin credentials and writes them to Vault.
+This would provide some level of separation of concerns, since managing admin credentials isn't necessarily part of the Project Syn bootstrap process.
+Additionally, having a separate tool allows us to have releases independent of the fairly complex Steward release process.
+
+==== Manage credentials by hand
+
+Another approach is to manage and renew the admin credentials by hand.
+
+== Decision
+
+Use service account tokens generated through the TokenRequest API and implement a custom controller to manage the service account, cluster role binding and tokens.
```

Comment:
So, I guess that the decision consist of one approach from "credential type" and one approach from "credential management", correct? 

We should make it clearer here, which approach for credential management is taken. I did not find out which one directly

Reply:
"implement a custom controller to manage the service account, cluster role binding and tokens" seems pretty clear to me

Reply:
yeah, my approach no deep expertise while reading is to ctrl+F and look for "custom controller" and would not find a match above, because there it's named "new controller".  So it would make it clearer, if above instead of "new controller" we would name it "new custom controller".  I'll add a proposal with links.

Reply:
https://github.com/appuio/openshift4-docs/pull/277/commits/2178830abc8dee1f493228cc26029adaa5d92a1f

Reply:
seems overkill for slightly more than a screen page of text, and the links don't even work

Reply:
true, they somehow do not work... now you changed the title, it should be clear...

Reply:
ah, that's how it works. not that obvious to find in https://docs.asciidoctor.org/asciidoc/latest/macros/xref/ within seconds :see_no_evil: 
thanks